src/Microsoft.ML.TimeSeries/Deseasonality/DeseasonalityBase.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Microsoft.ML.TimeSeries
+{
+    internal abstract class DeseasonalityBase
+    {
```

Comment:
This class is not providing any functionality in its base class. If all you are looking for is that derived classes should implement this function, should this be an interface instead?

Reply:
Changed to interface IDeseasonality accordingly

---
In reply to: [442523271](https://github.com/dotnet/machinelearning/pull/5202#discussion_r442523271) [](ancestors = 442523271)